### DISCUSSION TITLE
Unable to install Android universal apk on device

### DISCUSSION THREAD

**Discussion by haleksandre** [tauri-apps/tauri#9872] [Q&A] [answered]
I'm currently trying to build an Android apk with Tauri 2 with a few plugins. The build (Android Platform 33) works fine & the apk successfully gets created. But I can't install the apk on my device (Samsung S21 Android 14), the installer shows a message 'App not installed as package appears to be invalid'.

I'm not sure how to go from here, I tried lowering the target from 33 to 32 but it causes build issues as some plugin requires API-level 33, I also tried to raise the target to 34, but to no avail.

The only thing I could spot in the building phase that could be causing an issue is `Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.`

```bash
[✔] Environment
    - OS: Debian 12 X64
    ✔ webkit2gtk-4.1: 2.44.2
    ✔ rsvg2: 2.54.7
    ✔ rustc: 1.77.2 (25ef9e3d8 2024-04-09)
    ✔ cargo: 1.77.2 (e52e36006 2024-03-26)
    ✔ rustup: 1.27.0 (bbb9276d2 2024-03-08)
    ✔ Rust toolchain: 1.77.2-x86_64-unknown-linux-gnu (default)
    - node: 20.13.0
    - pnpm: 9.1.2
    - npm: 10.5.2
    - bun: 1.1.7

[-] Packages
    - tauri [RUST]: 2.0.0-beta.20
    - tauri-build [RUST]: 2.0.0-beta.16
    - wry [RUST]: 0.40.0
    - tao [RUST]: 0.28.0
    - @tauri-apps/api [NPM]: 2.0.0-beta.11
    - @tauri-apps/cli [NPM]: 2.0.0-beta.15

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:5173/
    - framework: Svelte
    - bundler: Vite
```

Does anyone know how to fix this or how to troubleshoot this further?

**Comment by FabianLars** [comment, marked as answer]
> the installer shows a message 'App not installed as package appears to be invalid'.

This typically means that you didn't sign the app, which is required to install the app. We don't have a guide in place yet, but you can check this out: https://github.com/tauri-apps/tauri-docs/issues/1674

**Reply by haleksandre** [reply]
You are correct. I misunderstood the Android 'unknown sources' option, I was under the impression that this would allow unsigned apks, but it doesn't. Once I've signed the apk, I was able to install it successfully.  Thanks for the help!
